gateway/api.go

```diff
@@ -965,34 +966,44 @@ func (gw *Gateway) handleGetAPIOAS(apiID string, modePublic bool) (interface{},
 
 }
 
-func (gw *Gateway) handleAddOrUpdateApi(apiID string, r *http.Request, fs afero.Fs, oasTyped bool) (interface{}, int) {
+func (gw *Gateway) handleAddOrUpdateApi(apiID string, r *http.Request, fs afero.Fs, oasAPI bool) (interface{}, int) {
 	if gw.GetConfig().UseDBAppConfigs {
 		log.Error("Rejected new API Definition due to UseDBAppConfigs = true")
 		return apiError("Due to enabled use_db_app_configs, please use the Dashboard API"), http.StatusInternalServerError
 	}
 
 	var newDef apidef.APIDefinition
-	var oas oas.OAS
+	var oasObj oas.OAS
+
+	isCreate := r.Method == http.MethodPost
```

Comment:
instead of checking every time, we can split handlers like `handleAddAPI` and `handleUpdateAPI`